### PR TITLE
Final API Programmer's Guide Engineering Changes Pre-launch

### LINKED ISSUE #9579: Final API Programmer's Guide Engineering Changes Pre-launch
## Problem statement
Before launching the new web-based Programmer's Guide, we want to perform one final check to ensure that any bugs/enhancements that need to be fixed before rolling out the new Programmer's Guide are captured.

Additionally, since the API Programmer's Guide was built in sections in parallel, checking the internal reference links (that link within the pages/sections) could not be completed at review stage. Therefore, we need to check those links to ensure they are all working.

## What you need to know

- [New Programmer's Guide](https://reportstream.cdc.gov/resources/reportstream-api)

## Issue/Bug/Enhancement list
**Needed Before Launch/Roll-out:**

- [x] Side navigation: Remove subpage links & jumplinks from all of the sections, except for the Documentation section, see the [Figma](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?type=design&node-id=4812-37084&scaling=min-zoom&page-id=3592%3A36210&starting-point-node-id=4812%3A37084) for reference - JH
- [x] Side navigation: Too late to make the [side nav sticky as you scroll](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?type=design&node-id=3592-36835&scaling=min-zoom&page-id=3592%3A36210&starting-point-node-id=4812%3A37084)? Can be future improvement if so - JH
  - Joseph Checking, may have to move out of MVP 
- [x] API Page and Documentation subpages: Has wider page/grid width than Getting Started and Documentation landing page
- [x] API Page: Card Links to Getting Started and Documentation (at top of page) link to non-existent pages
- [x] API Page: Download PDF link is broken - JH
- [x] Getting Started: Public key tool link is linking to the File Validator, should link to `https://reportstream.cdc.gov/resources/manage-public-key` - JH
- [x] API Page: "learn more" and "complete authentication" does not link to anything - AR
- [x] Getting started: Fake data file link does not work-AR
- [x] Getting started: Data model link in card box does not work -AR
- [x] Documentation: The "on this page" links do not correctly link  -AR
- [x] Documentation: All of the links in the sample payloads and outputs card do not work - AR
- [x] Getting Started: Above the file validator box, the new content is "When you’ve formatted your fake (non-PII) data file, test your data model using ReportStream’s file validation tool, and correct any errors you receive." You can see this change reflected in the [google doc](https://docs.google.com/document/d/1xsxkr7EnnpbXo8CcfdDcUnIYNKuFzY1u/edit#) as well (highlighted for easy reference right now).
- [x] Getting Started: Authentication section - Method 1, step 1, point 2 should now read the below content (You can see this change reflected in the [google doc](https://docs.google.com/document/d/1xsxkr7EnnpbXo8CcfdDcUnIYNKuFzY1u/edit#) as well - highlighted for easy reference right now).

> 2. Send the public key to the ReportStream team using our public key tool. Note: you’ll need to login to use that feature. If you do not have a login contact ReportStream support at reportstream@cdc.gov. ReportStream will associate the key with your configuration within ReportStream.
> 
> You only need to do this step once, not every time you submit reports. If you need to change your keys at any time, contact ReportStream support.

**Can Be Ticketed for future improvement:**
- [ ] All pages: Consistent vertical spacing between the nav and the page content/nav with the rest of the site (ex. the `product/overview`) - JH (added to #9688)
  - Difference exists for reason, needs review beyond MVP as a site-wide change. Will make a follow-on item post MVP. - CK and JA
- [ ] API Page: Comp match equidistant spacing between Getting started and Documentation cards, line rule, and Onboarding overview header (added to #9688)
- [ ] API Page: Make line rules thinner by ~.5x (Penny was also working on this on the Data Dashboard) -- this is the Horizontal Line thickness (should be thinner) (added to #9688)
- [ ] API Page: Replace alerts with Tip Callout component (used in File Validator & Public Key Tool, see Etan) (added to #9689)
- [ ] Getting Started Page: Comp match spacing between Data model card and following <p> (added to #9688)
- [ ] Getting Started Page: Comp match spacing between line rules and text (added to #9688)
- [ ] Documentation/Data page: Candidate for implementing sticky header (see Etan) (added to #9690)
- [ ] Getting started: We had notes that in the sample code snippets "healthy labs" was always styled blue, but one a couple instances are not blue -AR (added to #8941)

## Acceptance criteria
- [ ] All internal referencing links should be checked to ensure they are working properly.
- [ ] All issues/bugs/enhancements listed above are complete or ticketed for following sprints

## To do
- [x] Joseph to push main branch to trial front end after merging all existing PRs.
- [x] Design team to review newly deployed Programmer's Guide and note items that need changing before roll-out

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #9579

This PR:
- fixes some content per design
- sets the api guide sidenav as sticky
- fixes some app links
- updates USLink to not route `/assets` urls through react-router
- remove sidenav sublinks except for Documentation
- adds `ScrollRestoration` component to support hash links
- Updates USNavLink to pass on props
- Fixes sidenav index link with `end` prop

### CODE REVIEW

**Review by penny-lischer — APPROVED** [review]
lgtm

**Comment by jillian-hammer** [comment]
hi @jpandersen87 thank you so much, this looks great! That sticky side nav looks especially amazing. 

My notes are re: stubborn broken links:

- [ ] On `api/getting-started`: The data model card link is broken
- [ ] On `api/documentation`: None of the bulleted jump links are working
- [ ] On `api/documentation/data-model`: Ask on Entry jump link not working
- [ ] On `api/documentation/data-model`: Report and ordering facility elements jump link not working
- [ ] On `api/documentation/sample-payloads-and-output`: Sample HL7 2.5.1 payload and output jump link not working

File: frontend-react/src/App.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { toRelativeUrl } from "@okta/okta-auth-js";
 import { useIdleTimer } from "react-idle-timer";
 import { Suspense, useEffect, useRef } from "react";
 import { NetworkErrorBoundary } from "rest-hooks";
-import { useNavigate } from "react-router-dom";
+import { ScrollRestoration, useNavigate } from "react-router-dom";
```
Comment (etanb) [inline frontend-react/src/App.tsx:5]: Is this supposed to restore a users scroll position upon page reload? I don't see it working when I reload a page with the example of: `http://localhost:3000/resources/api/documentation/data-model#order-and-result-data-elements`

Comment (jpandersen87) [reply frontend-react/src/App.tsx]: 🤷‍♂️ Its react-router's official component that you're supposed to use, but like everything from that library its up in the air whether it'll work right.

**Review by penny-lischer — APPROVED** [review]
LGTM

**Review by etanb — APPROVED** [review]
LGTM, although `ScrollRestoration` doesn't seem to be doing anything for me 🤷

**Pull request merged**